parsers/rhea/rhea.go

```diff
@@ -0,0 +1,742 @@
+					Citations:            description.CitationStrings(),
+					Substrates:           description.SubstrateStrings(),
+					Products:             description.ProductStrings(),
+					SubstrateOrProducts:  description.SubstrateOrProductStrings(),
+					Location:             description.Location.Resource}
+				rhea.Reactions = append(rhea.Reactions, newReaction)
+			case "http://rdf.rhea-db.org/SmallMolecule", "http://rdf.rhea-db.org/Polymer":
+				compoundType := subclass.Resource[23:]
+				newCompound := Compound{
+					ID:        description.ID,
+					Accession: description.About,
+					Position:  description.Position,
+					Name:      description.Name,
+					HTMLName:  description.HTMLName,
+					Formula:   description.Formula,
+					Charge:    description.Charge,
+					Chebi:     description.Chebi.Resource,
+
+					CompoundID:        description.ID,
+					CompoundAccession: description.Accession,
+					CompoundName:      description.Name,
+					CompoundHTMLName:  description.HTMLName,
+					CompoundType:      compoundType}
+				if compoundType == "Polymer" {
+					newCompound.Chebi = description.UnderlyingChebi.Resource
+				}
+				// Add subclass Chebi
+				for _, sc := range description.Subclass {
+					if strings.Contains(sc.Resource, "CHEBI") {
+						newCompound.SubclassOfChebi = sc.Resource
+					}
+				}
+				// Add new reactive parts and new compounds to rhea
+				rhea.Compounds = append(rhea.Compounds, newCompound)
+			case "http://rdf.rhea-db.org/GenericPolypeptide", "http://rdf.rhea-db.org/GenericPolynucleotide", "http://rdf.rhea-db.org/GenericHeteropolysaccharide":
+				compoundType := subclass.Resource[23:]
+				newCompound := Compound{
+					Accession:        description.About,
+					CompoundID:       description.ID,
+					CompoundName:     description.Name,
+					CompoundHTMLName: description.HTMLName,
+					CompoundType:     compoundType}
+				compoundMap[description.About] = newCompound
+				compoundParticipantMap[description.ReactivePartXML.Resource] = description.About
+			}
+		}
+	}
+
+	// Go back and get the ReactiveParts
+	for _, description := range rdf.Descriptions {
+		for _, subclass := range description.Subclass {
+			switch subclass.Resource {
+			case "http://rdf.rhea-db.org/ReactivePart":
+				newCompound, ok := compoundMap[compoundParticipantMap[description.About]]
+				if ok != true {
+					return Rhea{}, errors.New("Could not find " + description.About)
+				}
+				newCompound.ID = description.ID
+				newCompound.CompoundAccession = description.About
+				newCompound.Position = description.Position
+				newCompound.Name = description.Name
+				newCompound.HTMLName = description.HTMLName
+				newCompound.Formula = description.Formula
+				newCompound.Charge = description.Charge
+				newCompound.Chebi = description.Chebi.Resource
+				rhea.Compounds = append(rhea.Compounds, newCompound)
+			}
+		}
+		for _, containsx := range description.ContainsX {
+			if strings.Contains(containsx.XMLName.Local, "contains") {
```

Comment:
```suggestion
			if !strings.Contains(containsx.XMLName.Local, "contains") {
			  continue;
			}
			
```
less nesting

Reply:
Interesting. Didn't consider doing that. Adding.